specs/ChainSpec/Upgradability.md

```diff
@@ -0,0 +1,71 @@
+# Upgradability
+
+This part of specification describes specifics of upgrading the protocol, and touches on few different parts of the system.
+
+High level upgradability consists of three parts:
+1. Updating without any changes to underlaying data structures or protocol;
+2. Updating when underlaying data structures changed (config, database or something else internal to the node and probably client specific);
+3. Updating with protocol changes that all validating nodes must adjust to.
+
+## Versioning
+
+There are 2 different important versions:
+- Version of binary defines it's internal data structures / database and configs. This version is client specific and doesn't need to be matching between nodes.
+- Version of the protocol, defining the "language" nodes are speaking.
+
+```rust
+/// Latest version of protocol that this binary can work with.
+type ProtocolVersion = u32;
+```
+
+Open questions: do we want to use something better than a number of versioning protocol?
```

Comment:
One another is YYYYMMDD timestamp, it's more human readable and imply the date of when this version start. it's also strictly increased.

Reply:
Evgeny Kapun mentioned a protocol version changed need to be tested on testnet first, and YYYYMMDD doesn't reasoning good in that case. Also it doesn't give much benefit over number so it's not good.